rebasehelper/checkers/abipkgdiff_tool.py

```diff
@@ -112,17 +119,98 @@ def run_check(cls, results_dir, **kwargs):
+            Example abipkgdiff line:
+            '  Functions changes summary: 3 Removed, 0 Changed, 0 Added functions (4 filtered out)'
+            """
+            summary_re = re.compile(r'''^
+            \s+(?P<kind>[\w\s]+changes\s+summary):\s+
+            (?P<changes>.+)
+            $
+            ''', re.VERBOSE)
+            changes_re = re.compile(r'''
+            (?P<count>\d+)\s+
+            (?P<what>Added|Changed|Removed)(\s+functions|variables)?
+            (\s+\((?P<filtered_out>\d+)\s+filtered\s+out\))?
+            ''', re.VERBOSE)
+            result = {}
+            for line in lines:
+                ms = summary_re.match(line)
+                if ms:
+                    ds = ms.groupdict()
+                    result[ds['kind']] = {}
+                    for mc in changes_re.finditer(ds['changes']):
+                        dc = mc.groupdict()
+                        if int(dc['count']) or dc['filtered_out']:
+                            result[ds['kind']][dc['what']] = {'count': dc['count'], 'filtered_out': dc['filtered_out']}
+            return result
+
+        pkgs = {}
+        for pkg, ret_code in six.iteritems(reports):
+            # If no abi changes for the package, store empty dictionary
+            cur_pkg = {}
+            if ret_code:
+                with open(os.path.join(cls.results_dir, pkg + '.txt'), 'r') as f:
+                    cur_pkg = parse_changes(f.readlines())
+                    cls.abi_changes = True
+            pkgs[pkg] = cur_pkg
+        return pkgs
+
+    @classmethod
+    def format(cls, data):
+        """
+        Format abipkgdiff output
+        :param data: abipkgdiff dictionary output
+        :return: formated abipkgdiff list of strings
+        """
+        output_lines = [cls.get_underlined_title('abipkgdiff')]
+        if not data['abi_changes']:
+            output_lines.append('No ABI changes occured')
+            return output_lines
+
+        for pkg_name, pkg_changes in sorted(six.iteritems(data['packages'])):
+            if not pkg_changes:
+                continue
+            output_lines.append("ABI changes in %s:" % pkg_name)
+
+            for sum_title, changes_list in sorted(six.iteritems(pkg_changes)):
+                if not changes_list:
+                    continue
+                output_lines.append("%s" % sum_title)
+
+                for change_name, change_info in sorted(six.iteritems(changes_list)):
+                    if change_info['filtered_out']:
+                        output_lines.append(" - {} {} (filtered out {})".format(change_name,
+                                                                                 change_info['count'],
+                                                                                 change_info['filtered_out']))
+                    else:
+                        output_lines.append(" - {} {}".format(change_name, change_info['count']))
+        output_lines.append("Details in %s:" % data['path'])
```

Comment:
You shouldn't mix `%` and `format()`. I actually think we should start to enforce using only `format()` in rebase-helper, `%` is going to be [removed](https://docs.python.org/release/3.1.5/library/stdtypes.html#old-string-formatting-operations) from Python in the future anyway.